### PR TITLE
(OSD-25580) Ship Network Live Migration Metrics to Telemetry

### PR DESCRIPTION
**Pull request by dakotalongRH**
Introduce new alerting and metrics for the SDN to OVN migration

### What type of PR is this?
_(bug/feature/cleanup/documentation)_

### Which Jira/Github issue(s) this PR fixes?
https://issues.redhat.com/browse/OSD-25580
https://issues.redhat.com/browse/OSD-25335

_Fixes #_

### Special notes for your reviewer:
Note, the expressions used for checking migration are taken from [OSD-25335](https://issues.redhat.com//browse/OSD-25335).

### Pre-checks (if applicable):
- [X] Tested latest changes against a cluster
- [ ] Included documentation changes with PR
- [ ] If this is a new object that is not intended for the FedRAMP environment (if unsure, please reach out to team FedRAMP), please exclude it with:

    ```yaml
    matchExpressions:
    - key: api.openshift.com/fedramp
      operator: NotIn
      values: ["true"]
    ```

### CODE REVIEW

File: deploy/sre-prometheus/100-network-live-migration.PrometheusRule.yaml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+                  timestamp(
+                    openshift_network_operator_live_migration_condition{type="NetworkTypeMigrationInProgress"} == 1
+                  )[5d:]
+                )
+              )
+            )
+          )
+          and on()
+          openshift_network_operator_live_migration_condition{type="NetworkTypeMigrationInProgress"} == 1
+        record: sre:network-live-migration:condition
```
Comment (abyrne55) [inline deploy/sre-prometheus/100-network-live-migration.PrometheusRule.yaml:35]: Do we know if `sre:...` is valid for getting this metric exported to Telemeter? The card/[CMO regex](https://github.com/openshift/cluster-monitoring-operator/blob/master/Documentation/data-collection.md?plain=1#L38) would seem to indicate that only `cluster:usage...` metrics get this treatment, but its very possible there's other components picking up `sre:...` metrics that I'm not aware of

Regardless, I wouldn't name this recording rule "condition", as I think it's just recording an integer number of seconds of how long the migration is/was in_progress. Perhaps "duration"?

**Review by abyrne55 — CHANGES_REQUESTED** [review]
Good work @dakotalongRH. In addition to my inline comments, could we add another recording rule that just exports `openshift_network_operator_live_migration_condition` as-is into an exported name (e.g., `cluster:usage:openshift_network_operator_live_migration_condition`)?

File: deploy/sre-prometheus/100-network-live-migration.PrometheusRule.yaml (hunk 1 above)
Comment (dakotalongRH) [reply deploy/sre-prometheus/100-network-live-migration.PrometheusRule.yaml:35]: Changing to "cluster:usage:network-live-migration:*"

**Review by abyrne55 — CHANGES_REQUESTED** [review]

File: deploy/sdn-ovn-migration/100-network-live-migration.PrometheusRule.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+                    openshift_network_operator_live_migration_condition{type="NetworkTypeMigrationInProgress"} == 1
+                  )[5d:]
+                )
+              )
+            )
+            and on()
+            openshift_network_operator_live_migration_condition{type="NetworkTypeMigrationInProgress"} == 1
+        record: cluster:usage:network_live_migration_duration
+      - expr: openshift_network_operator_live_migration_blocked
+        record: cluster:usage:network_live_migration_blocked
```
Comment (abyrne55) [inline deploy/sdn-ovn-migration/100-network-live-migration.PrometheusRule.yaml:35]:
```suggestion
      - expr: openshift_network_operator_live_migration_blocked
        record: cluster:usage:network_live_migration_blocked
      - expr: openshift_network_operator_live_migration_condition
        record: cluster:usage:network_live_migration_condition
```
Looking great! Just need one more addition. See my [earlier comment](https://github.com/openshift/managed-cluster-config/pull/2258#pullrequestreview-2410519297) for details

Comment (dakotalongRH) [reply deploy/sdn-ovn-migration/100-network-live-migration.PrometheusRule.yaml:35]: Done :)

File: deploy/sdn-ovn-migration/100-network-live-migration.PrometheusRule.yaml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+      - expr: openshift_network_operator_live_migration_condition
+        record: cluster:usage:network_live_migration_condition
+  - name: sre-network-live-migration-alerts
+    rules:
+      - alert: NetworkMigrationDelayedSRE
+        expr: (cluster:usage:network_live_migration_duration > (sum(cluster:node_roles) * 1800))
```
Comment (joshbranham) [inline deploy/sdn-ovn-migration/100-network-live-migration.PrometheusRule.yaml:41]: @abyrne55 @dakotalongRH should I expect to see `cluster:node_roles` in Telemeter? I am working on the SLO dashboard and I don't seem to be able to find that metric which makes me wonder if this expr works?

**Pull request merged**